sale_order_line_display_stock_per_warehouse/models/sale_order_line.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 Akretion (https://www.akretion.com).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    stock_per_warehouse_info = fields.Html(
+        string="Stock / Warehouse", compute="_compute_stock_per_warehouse_info"
+    )
+
+    @api.depends("product_id")
+    def _compute_stock_per_warehouse_info(self):
+        for line in self:
+            info = ""
+            if line.product_id:
+                warehouses = self.env["stock.warehouse"].search(
+                    [
+                        ("company_id", "=", line.order_id.company_id.id),
+                        ("display_stock_on_sol", "=", True),
+                    ]
+                )
+                for warehouse in warehouses:
+                    qty_data = (
+                        self.env["stock.quant"]
+                        .with_context(warehouse=warehouse.id)
+                        .read_group(
+                            domain=[
+                                ("product_id", "=", line.product_id.id),
+                                ("location_id", "child_of", warehouse.lot_stock_id.id),
+                            ],
+                            fields=["quantity:sum"],
+                            groupby=[],
+                        )[0]["quantity"]
+                    )
+                    qty_available = qty_data if qty_data else 0
```

Comment:
Make it simplier and correct. 
```suggestion
                    qty_available = line.product_id.with_context(warehouse=warehouse.id).qty_available
```

With the current implementation you have the quantity that you have physically in stock. What we need is the quantity that you can sell. The field qty_available depends on the context passed in the warehouse so it' simplier to reuse this field.

Reply:
thank you for your review, I fixed it.